core/src/main/scala/com/pingcap/tispark/utils/HttpClientUtil.scala

```diff
@@ -0,0 +1,243 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tispark.utils
+
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.pingcap.tikv.TiConfiguration
+import com.pingcap.tispark.TiConfigConst
+import org.apache.spark.sql.SparkSession
+import org.slf4j.LoggerFactory
+import scalaj.http.{Http, HttpOptions, HttpResponse}
+import java.io.{BufferedReader, FileInputStream, InputStream, InputStreamReader}
+import java.nio.charset.StandardCharsets
+import java.security.{KeyFactory, KeyStore, PrivateKey}
+import java.security.cert.{Certificate, CertificateFactory}
+import java.security.spec.PKCS8EncodedKeySpec
+import javax.net.ssl.{KeyManagerFactory, SSLContext, TrustManagerFactory}
+import java.util.stream.Collectors
+import java.util.Base64
+
+class HttpClientUtil {
+  private val logger = LoggerFactory.getLogger(getClass.getName)
+
+  /**
+   * HTTP POST
+   *
+   * @param url    server url
+   * @param msg    post entry object
+   * @param headers HTTP header
+   * @return HTTP response string
+   */
+  def postJSON(
+      url: String,
+      msg: Object,
+      headers: Map[String, String] = null): HttpResponse[String] = {
+    try {
+      val mapper = new ObjectMapper()
+        .registerModule(DefaultScalaModule)
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+      val msgString = mapper.writeValueAsString(msg)
+
+      val header: Map[String, String] =
+        if (headers != null) headers else Map("Content-Type" -> "application/json")
+
+      val resp = Http(url).postData(msgString).headers(header).asString
+      checkResp(url, resp)
+
+      resp
+    } catch {
+      case e: Throwable =>
+        throw e
+    }
+  }
+
+  /**
+   * HTTP GET
+   *
+   * @param url      server url
+   * @param headers  HTTP header
+   * @return HTTP response string
+   */
+  def get(url: String, headers: Map[String, String] = null): HttpResponse[String] = {
+    try {
+      val header: Map[String, String] =
+        if (headers != null) headers else Map("Content-Type" -> "application/json")
+
+      val resp = Http(url).headers(header).asString
+      checkResp(url, resp)
+
+      resp
+    } catch {
+      case e: Throwable =>
+        throw e
+    }
+  }
+
+  /**
+   * HTTPS GET
+   *
+   * Support PKCS#8 and JKS format.
+   *
+   * @param url      server url
+   * @param conf     which contain certificate path
+   * @param headers  HTTP header
+   * @return HTTP response string
+   */
+  def getHttps(
+      url: String,
+      conf: TiConfiguration,
+      headers: Map[String, String] = null): HttpResponse[String] = {
+    try {
+      val header: Map[String, String] =
+        if (headers != null) headers else Map("Content-Type" -> "application/json")
```

Comment:
why application/json type?

Reply:
`application/json` type tells the server that the client comes with JSON content. However it is a GET request here, we don't need to add this head here. The `application/json` head would not affect in GET here.

Reply:
I delete default head ("Content-Type" -> "application/json") in GET.